### PR TITLE
chanfunding: Add/improve tests for CoinSelectSubtractFees

### PR DESCRIPTION
**Pull request by bjarnemagnussen**
It seems that the dust-output and high-fee test cases for `CoinSelectSubtractFees` were not independently testing the code as the one triggered the other and vice versa due to testing with low fee values that gave rise to dust outputs (see also comments for the commits).

Also a new test case `"spend with change"` was added to check that a non-zero change output can be created.

#### Pull Request Checklist

- [x] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
- [x] All changes are Go version 1.12 compliant
- [x] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
- [x] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [x] For bug fixes: Code is accompanied by new tests which trigger
  the bug being fixed to prevent regressions
- [x] Any new logging statements use an appropriate subsystem and
  logging level
- [x] Code has been formatted with `go fmt`
- [x] Protobuf files (`lnrpc/**/*.proto`) have been formatted with
  `make rpc-format` and compiled with `make rpc`
- [x] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [x] Running `make check` does not fail any tests
- [x] Running `go vet` does not report any issues
- [x] Running `make lint` does not report any **new** issues that did not
  already exist
- [x] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.
- [x] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)

### CODE REVIEW

**Comment by bjarnemagnussen** [comment]
I have rebased to current master (although that was unnecessary) and added error-string matching to the tests of `CoinSelectSubtractFees`.

I also checked to see if it would make sense to add it to `CoinSelect`. But an error there is only possible from demanding a too high amount when selecting inputs and therefore I left it unchanged.

File: lnwallet/chanfunding/coin_select_test.go [hunk 1]
```diff
@@ -209,6 +210,12 @@ func TestCoinSelectSubtractFees(t *testing.T) {
 	const dustLimit = btcutil.Amount(1000)
 	const dust = btcutil.Amount(100)
 
+	// removeAmounts replaces any amounts in string with "<amt>".
+	removeAmounts := func(s string) string {
```
Comment (halseth) [inline lnwallet/chanfunding/coin_select_test.go:214]: magic

File: lnwallet/chanfunding/coin_select_test.go [hunk 2]
```diff
@@ -371,19 +379,26 @@ func TestCoinSelectSubtractFees(t *testing.T) {
...
 			)
-			if !test.expectErr && err != nil {
-				t.Fatalf(err.Error())
+			if err != nil {
+				switch {
+				case test.expectErr == "":
+					t.Fatalf(err.Error())
+
+				case test.expectErr != removeAmounts(err.Error()):
+					t.Fatalf("expected different error")
```
Comment (halseth) [inline lnwallet/chanfunding/coin_select_test.go:388]: add expected and actual error to the string here

File: lnwallet/chanfunding/coin_select_test.go [hunk 3]
```diff
@@ -371,19 +379,26 @@ func TestCoinSelectSubtractFees(t *testing.T) {
...
+
+				default:
+					// If we got an expected error, there is
```
Comment (halseth) [inline lnwallet/chanfunding/coin_select_test.go:391]: nit: move comment above case

**Review by halseth — APPROVED** [review]
Awesome change, thanks! Only a few nits from me, otherwise LGTM 👍

**Comment by bjarnemagnussen** [comment]
Thanks!

**Comment by Roasbeef** [comment]
Is this blocked on anything?

**Comment by halseth** [comment]
> Is this blocked on anything?

No.

@bjarnemagnussen Needs rebase to trigger the new build checks, then we can merge this!

**Comment by bjarnemagnussen** [comment]
Rebased!
